stripe/src/main/java/com/stripe/android/IssuingCardPinService.java

```diff
@@ -0,0 +1,290 @@
+            String verificationId = (String) arguments.get(ARGUMENT_VERIFICATION_ID);
+            String userOneTimeCode = (String) arguments.get(ARGUMENT_ONE_TIME_CODE);
+            IssuingCardPinRetrievalListener listener = mRetrievalListeners.get(operationId);
+
+            try {
+                String pin = mApiHandler.retrieveIssuingCardPin(
+                        cardId,
+                        verificationId,
+                        userOneTimeCode,
+                        ephemeralKey.getSecret());
+
+                if (listener != null) {
+                    listener.onIssuingCardPinRetrieved(pin);
+                }
+            } catch (InvalidRequestException e) {
+                if (listener != null) {
+                    if ("expired".equals(e.getErrorCode())) {
+                        listener.onError(
+                                CardPinActionError.ONE_TIME_CODE_EXPIRED,
+                                "The one-time code has expired",
+                                null);
+                    }
+                    if ("incorrect_code".equals(e.getErrorCode())) {
+                        listener.onError(
+                                CardPinActionError.ONE_TIME_CODE_INCORRECT,
+                                "The one-time code was incorrect",
+                                null);
+                    }
+                    if ("too_many_attempts".equals(e.getErrorCode())) {
+                        listener.onError(
+                                CardPinActionError.ONE_TIME_CODE_TOO_MANY_ATTEMPTS,
+                                "The verification challenge was attempted too many times",
+                                null);
+                    } else {
+                        listener.onError(
+                                CardPinActionError.UNKNOWN_ERROR,
+                                "An error occurred retrieving the PIN",
+                                e);
+                    }
+                }
+            } catch (APIConnectionException |
+                    APIException |
+                    AuthenticationException |
+                    JSONException |
+                    CardException e) {
+                if (listener != null) {
+                    listener.onError(
+                            CardPinActionError.UNKNOWN_ERROR,
+                            "An error occurred retrieving the PIN",
+                            e);
+                }
+            }
+        }
+        if (PIN_UPDATE.equals(action)) {
+            String cardId = (String) arguments.get(ARGUMENT_CARD_ID);
+            String newPin = (String) arguments.get(ARGUMENT_NEW_PIN);
+            String verificationId = (String) arguments.get(ARGUMENT_VERIFICATION_ID);
+            String userOneTimeCode = (String) arguments.get(ARGUMENT_ONE_TIME_CODE);
+            IssuingCardPinUpdateListener listener = mUpdateListeners.get(operationId);
```

Comment:
let's use `remove` instead of `get` so that the listener is removed after the callback

Reply:
👍